Test

patrick-luethi opened the pull request:
test? :)

patrick-luethi committed:
Add a very basic strategy choice to the ModelRenderer

patrick-luethi committed:
Change naming from render_strategy to visualization_choice_strategy

patrick-luethi committed:
Replace exception with assert

patrick-luethi committed:
Change store of visualization choice strategy from constant to field

patrick-luethi committed:
Copy the way purpose is stored to store the semantic zoom level alike

dimitar-asenov committed:
Enable the rendering of all items when rendering a screenshot

dimitar-asenov commented:
Alright, let's see how this goes.

patrick-luethi committed:
Add functionality to register a semantic zoom level in the ModelRenderer

patrick-luethi committed:
Add functionality to register visualizations for a semantic zoom level

patrick-luethi committed:
Update the RootItem class to set the default semantic zoom level

patrick-luethi committed:
Extended the ModelRenderer to consider semantic zoom level when choosing a visualization

patrick-luethi committed:
Add a handler to change the semantic zoom level by pressing F4

dimitar-asenov committed:
Unify two almost identical methods into one

dimitar-asenov committed:
Move inline private method implementation in cpp file

dimitar-asenov committed:
Use two independent constraint solvers - vertical and horizontal

This will be used to improve performance by making each solver cache
some data.

dimitar-asenov committed:
Reorder constraint solver API calls according to documentation

dimitar-asenov committed:
Reuse the LP data structure when solving anchor layout constraints.

Speeds up things quite a bit, as no memory allocations are needed and a
lot of things are preinitialized.

dimitar-asenov committed:
Improve text performance: use QStaticText.size instead of QFontMetrics

patrick-luethi committed:
Add class visualization of the public interface only semantic zoom level

patrick-luethi committed:
Register public interface only semantic zoom level

patrick-luethi committed:
Fix codestyle line length

patrick-luethi committed:
Encapsulate visualization groups management in separate class

dimitar-asenov committed:
Prepare the expression editor for supporting sub expressions

dimitar-asenov committed:
Add a few more error test cases to the expression parsing

dimitar-asenov committed:
Group expressions surrounded by [],{}, or () to improve performance

This comes at a possibly less useful parsing when the expressions are
incorrect.

dimitar-asenov committed:
Group expressions surrounded by [],{}, or () to improve performance

This comes at a possibly less useful parsing when the expressions are
incorrect.

dimitar-asenov committed:
Add one more error test case for expression parsing

dimitar-asenov committed:
Handle more cases of erroneous expressions including sub-expressions

patrick-luethi committed:
Use new way of specifying styles in VClassSzPublic

patrick-luethi committed:
Fix copyright year

patrick-luethi committed:
Fix codestyle

patrick-luethi committed:
Change layout of methods in the public interface semantic zoom level

patrick-luethi committed:
Add first method visualization for public interface semantic zoom level

patrick-luethi committed:
Register method visualization for the public interface semantic zoomm level

patrick-luethi committed:
Clean up some unused code and declarations

dimitar-asenov committed:
Fix the scaling of the command prompt when making screenshots

dimitar-asenov committed:
Use TAB for adding a new list element.

dimitar-asenov committed:
Add a handler for array initializers to make it easier to edit matrices

dimitar-asenov committed:
Fix the visualization of list item backgrounds to respect item order

dimitar-asenov committed:
Automatically create a default list element when a letter is pressed

The pressed letter is sent as an event to the newly created element

dimitar-asenov committed:
Automatically show a command prompt when an unhandled letter is pressed

dimitar-asenov committed:
Fix an incorrect condition when creating list elements with TAB

dimitar-asenov committed:
Revert the automatic prompt invocation on unhadled letter

It turns out, this is a bad idea, especially the way the expression
editing works.

dimitar-asenov committed:
Make it possible to add empty dimensions in new expressions.

dimitar-asenov committed:
Delete unused objects

dimitar-asenov committed:
Improve the parser's speed by pruning trees with bad parse results

dimitar-asenov committed:
Add an assertion

dimitar-asenov committed:
Change the symbol for the *= operator to an icon

dimitar-asenov committed:
Fix the ControlFlow method style to not have an outline

patrick-luethi committed:
Fix copyright year

patrick-luethi committed:
Revert public interface semantic zoom class visualization to look more like the default visualization

patrick-luethi committed:
Revert the public interface semantic zoom method visualization to look more like the default visualization

patrick-luethi committed:
Add visualization choice strategy type over semantic zoom over purpose

patrick-luethi committed:
Fix copyright year

patrick-luethi committed:
Change default visualization choice strategy

patrick-luethi committed:
Add functions to retrieve registered purpose- and semantic zoom level-ids by name

patrick-luethi committed:
Add functions to register visualizations providing the name of the purpose or semantic zoom level it should be registered for

patrick-luethi committed:
Update visualization registration to use new registration by name

patrick-luethi committed:
Fix some graphic issues when scaling items

patrick-luethi committed:
Remove some visualizations from the public interface semantic zoom visualization

patrick-luethi committed:
Change position layout to make it try expanding children

patrick-luethi committed:
Change public interface semantic zoom class visualization to scale itself according to the space available

patrick-luethi committed:
Update MainView to update the scene after zooming

patrick-luethi committed:
Fix code style

patrick-luethi committed:
Move function to a related section in the code

dimitar-asenov committed:
Make it possible to select an icon and press DEL to delete the object

dimitar-asenov committed:
Add some convenience functions to the log.

dimitar-asenov committed:
Improve the logging setup in ModelBase

dimitar-asenov committed:
Improve the logging setup in InteractionBase

dimitar-asenov committed:
Add debug logs to reference resolution.

dimitar-asenov committed:
Introduce a helper class for profiling performance

dimitar-asenov committed:
Add some profiler calls

patrick-luethi committed:
Introduce naming convention for purpose- and semantic zoom level registrations

patrick-luethi committed:
Change name of getTotalScale to totalScale in Item and moved it's body to source

patrick-luethi committed:
Change position layout to not interfere with the stretchable property

patrick-luethi committed:
Add high level semantic zoom visualization

patrick-luethi committed:
Register high level semantic zoom visualization

dimitar-asenov committed:
Show auto completion prompts in events to avoid update issues

dimitar-asenov committed:
Fix Scene's destructor to correctly delete children

The old version deletes the same object multiple times when there are
multiple top-level objects.

dimitar-asenov commented:
BTW, Keep the length of the first line of each commit title to at most 72 characters. This way they are displayed as good as possible in many git tools, including github.

dimitar-asenov committed:
Introduce a helper method in Item - mainViewScalingFactor()

It returns the current geometric zoom level of the main view associated
with the item's scene

dimitar-asenov committed:
Add a feature to update the geometry of items on zoom.

patrick-luethi committed:
Use new way of retrieving the mainview's scaling factor and test the rebase

patrick-luethi committed:
Update PositionLayout to use new feature to be notified on zoom change

dimitar-asenov committed:
Adjust the copyright year

dimitar-asenov committed:
Fix operations to make them work with correct sizes (local/parent/scene)

dimitar-asenov committed:
Add some convenience options to move the cursor when using moveCursor()

patrick-luethi committed:
Remove recursive method of duplicate feature totalScale

patrick-luethi committed:
Clean up strategy choice by reducing redundancy

patrick-luethi committed:
Add individual semantic zoom level switching

patrick-luethi committed:
Add setItemScale to constraint size of child items after updating their parent's scale

patrick-luethi committed:
Update high level semantic zoom level visualization

patrick-luethi committed:
Clean up style file for ModuleSz

patrick-luethi committed:
Update scaling algorithm in PositionLayout

patrick-luethi committed:
Refactor PositionLayout's updateGeometry method

patrick-luethi committed:
Use different scaling algorithm

dimitar-asenov committed:
Properly update what items must be updated on zoom

patrick-luethi committed:
Restructure body of position layout's updateGeometry method

patrick-luethi committed:
Remove unnecessary setItemScale calls in during item scaling in position layout

dimitar-asenov committed:
Let the renderer render method bodies, don't use a fixed class

dimitar-asenov committed:
Remove the no longer necessary VMethodCF class

dimitar-asenov committed:
Remove unused headers

dimitar-asenov committed:
Use the declarative framework for VProject and VModule

Instead of using PanelBorderLayout

dimitar-asenov committed:
Use DeclarativeItem for defining VCatchClause

patrick-luethi committed:
Add declaration visualization

patrick-luethi committed:
Register declaration visualization tests

patrick-luethi committed:
Add semantic zoom level ordering manager

patrick-luethi committed:
Integrate semantic zoom level ordering

dimitar-asenov committed:
Add an EmptyItem visualization class

dimitar-asenov committed:
Use the declarative framework for the visualization of try-catch-finally

dimitar-asenov committed:
Use the declarative framework for VForEachStatement

dimitar-asenov committed:
Fix a bug where -1 for purpose would be returned for parentless items

dimitar-asenov committed:
Use the declarative framework to define VComposite

dimitar-asenov committed:
Remove PanelLayout and PanelBorderLayout

dimitar-asenov committed:
Fix the default of several visualizations to be a non-null object

dimitar-asenov committed:
Add convenience methods to create item elements in a declarative item

dimitar-asenov committed:
Use the simpler methods for defining item elements in VProject forms

dimitar-asenov committed:
Change many styles to use the new Property<> class

The only remaining styles that still use the old loading mechanism are
in OOVisualization.

dimitar-asenov committed:
Use Super<> when inheriting styles

dimitar-asenov committed:
Use Property<> in all styles in VisualizationBase

Two classes were missed during the last refactoring.

patrick-luethi committed:
Remove reset scale

patrick-luethi committed:
Change setItemScale to only scale if the parent is a PositionLayout

patrick-luethi committed:
Add threshold for scale change to take effect

patrick-luethi committed:
Change name of parameter of setItemScale

patrick-luethi committed:
Change declaration semantic zoom visualization

patrick-luethi committed:
Fix bug in individual zoom changing

dimitar-asenov committed:
Remove an used class

dimitar-asenov committed:
Make reference targets a non-undo property and clean up reference design

This resulted in a lot of const modifier propagation.

dimitar-asenov committed:
Fix a memory leak.

dimitar-asenov committed:
Fix a double delete bug and a related memory leak

dimitar-asenov committed:
Improve reference resolution code and move it to the Reference class.

dimitar-asenov committed:
Fix the stretch factor of the content elements of VModule and VProject

dimitar-asenov committed:
Make DeclarativeItem::item<> be usable in more cases.

Specifically the new version works when, a visualization for a base type
is used to show a derived node type.

dimitar-asenov committed:
Add subtyping queries based on the string identifier of types.

dimitar-asenov committed:
Add type checks to CompositeNode

dimitar-asenov committed:
Add a dedicated NameText node type for object name definitions

dimitar-asenov committed:
Clean up some unused includes and references to Text.

dimitar-asenov committed:
Add a basic mechanism that resolves reference targets on model changes

There are still a lot of things to do, such as make sure that including
name imports causes updates.

dimitar-asenov committed:
Fix a redundant shape update resulting is a massive scene repaint.

dimitar-asenov committed:
Skip reference exploration when there is nothing to resolve.

dimitar-asenov committed:
Keep the model cached as a field in each Node instance.

dimitar-asenov committed:
Optimize reference unresolution.

A factor of 10 improvement.

dimitar-asenov committed:
Add unresolution for NameImports.

dimitar-asenov committed:
Improve reference resolution logging.

patrick-luethi committed:
Improve performance of VisualizationGroupsManager

patrick-luethi committed:
Fix icon size for project abstraction visualization

patrick-luethi committed:
Introduce header scaling for projects, modules and declaration abstractions

patrick-luethi committed:
Remove global semantic zoom level change pressing F keys

patrick-luethi committed:
Remove individual semantic zoom change using F keys

patrick-luethi committed:
Remove public interface abstraction registration

patrick-luethi committed:
Remove obsolete dependency on PanelBorderLayout

patrick-luethi committed:
Put semantic zoom level switching on alt+mousewheel

patrick-luethi committed:
Fix identation

patrick-luethi committed:
Fix propagation of mouse wheel event

patrick-luethi committed:
Extend scene to also have a previous scale factor property

patrick-luethi committed:
Add getter for previous main view scale to item

patrick-luethi committed:
Add full update handling to position layout

patrick-luethi committed:
Change geometric zoom step to be smaller

patrick-luethi committed:
Add semantic/geometric zoom mix feature

patrick-luethi committed:
Add comments to semantic zoom level change part of GenericHandler

patrick-luethi committed:
Remove unnecessary update

patrick-luethi committed:
Introduce set cursor event for global zoom change

patrick-luethi committed:
Add comment to basicStrategy

patrick-luethi committed:
Fix overscale issue in setItemScale

patrick-luethi committed:
Add ArrangementAlgorithmItem class to be used by the arrangment algorithm

patrick-luethi committed:
Clean up arrangment algorithm

patrick-luethi committed:
Add saving geometric scales for the automatic semantic zoom level change

patrick-luethi committed:
Fix bug related to switching the semantic zoom level

patrick-luethi committed:
Rename semantic zoom levels

patrick-luethi committed:
Correct automatic abstraction semantic zoom level

patrick-luethi committed:
Remove former high level semantic zoom level visualization

patrick-luethi committed:
Remove unused dependency

patrick-luethi committed:
Fix full update bug in PositionLayout

patrick-luethi committed:
Add automatic semantic zoom switch only if all children are abstracted

patrick-luethi committed:
Fix wheel event propagation

patrick-luethi committed:
Clean up code and comments

patrick-luethi committed:
Update PositionLayout determineChildren to only delete changed items

patrick-luethi committed:
Update automatic sz level switch to use perceived width threshold
